tests/unit/test_serialization.py

```diff
@@ -267,105 +375,167 @@ def assert_almost_equal(value: np.ndarray, reference: Any):
         np.testing.assert_almost_equal(value, reference)
 
 
-def assert_scenario_correct(
-    deserialized_dataset: Mapping[ComponentType, np.ndarray],
+def assert_single_dataset_entries(
+    deserialized_dataset: SingleDataset,
     serialized_dataset: Mapping[str, Any],
     sparse_components: list[ComponentType],
+    data_filter,
 ):
+    """Check a SingleDataset's individual components for correctness"""
+    # TODO What is the purpose of this check? Implement without using derserialized_dataset
     for key in serialized_dataset["data"]:
-        if key not in deserialized_dataset:
+        if key not in deserialized_dataset and not is_filtered_out(data_filter, key):
             assert key in sparse_components
+            assert False
 
-    for component, component_result in deserialized_dataset.items():
-        assert isinstance(component_result, np.ndarray)
-
-        component_input = serialized_dataset["data"][component]
-        assert len(component_result) == len(component_input)
-        for result_entry, input_entry in zip(component_result, component_input):
-            if isinstance(input_entry, dict):
-                for attr, values in input_entry.items():
-                    assert attr in result_entry.dtype.names
-                    assert_almost_equal(result_entry[attr], values)
+    for component, component_input in serialized_dataset["data"].items():
+        if is_filtered_out(data_filter, component):
+            assert component not in deserialized_dataset
+            continue
+
+        component_result = deserialized_dataset[component]
+        assert isinstance(component_input, list)  # TODO Why?
```

Comment:
I this the example you provided to differentiate compact list from non-compact helps understand why this check?
```
{
    "version": "1.0", 
    "type": dataset_type, 
    "is_batch": False,
    "attributes": {
        "sym_load": ["status", "p_specified", "q_specified"],
        "source": ["status"],
    }
    "data": {
        "line": [{}, {"from_status": 1, "to_status": 1}],
        "source": [[1], [0], {"status": 1, "u_ref": 1.03}],
        "sym_load": [[9, 1.01e6, 0.21e6], [10, 1.02e6, 0.22e6]],
}
```
If you see the components in data, everything is organized in a list. So I guess this is to check that the serialization format is correct.